MyApplication2/app/src/main/java/shuvalov/nikita/whatartinwalmart/MainActivity.java

```diff
@@ -0,0 +1,108 @@
+
+
+    //Search parameter is added in the form of %s to be used in String.format()
+    public static final String REQUEST_URL = "http://api.walmartlabs.com/v1/search?query=%s&format=json&apiKey=6we7gvkbu5sujb78rge8y5dy";
+
+    Button chocoButt, teaButt, cerealButt;
+    RecyclerView mRecyclerView;
+    RecyclerWalmartAdapter mAdapter;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        chocoButt= (Button)findViewById(R.id.chocolate_button);
+        teaButt= (Button)findViewById(R.id.tea_button);
+        cerealButt= (Button)findViewById(R.id.cereal_button);
+
+        View.OnClickListener onClickListener = new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                ConnectivityManager connectivityManager = (ConnectivityManager) getSystemService(CONNECTIVITY_SERVICE);
+                NetworkInfo info = connectivityManager.getActiveNetworkInfo();
+                String searchParam="";
+                if(info!=null && info.isConnected()){
+                    RequestQueue queue = Volley.newRequestQueue(view.getContext());
+                    switch (view.getId()){
+                        case R.id.cereal_button:
+                            searchParam="cereal";
+                            break;
+                        case R.id.chocolate_button:
+                            searchParam="chocolate";
+                            break;
+                        case R.id.tea_button:
+                            searchParam="tea";
+                            break;
+                    }
+                    StringRequest stringRequest = new StringRequest(Request.Method.GET, String.format(REQUEST_URL, searchParam), new com.android.volley.Response.Listener<String>() {
+                        @Override
+                        public void onResponse(String response) {
+                            ArrayList<WalmartItem> walmartItems = new ArrayList<>();
+                            try {
+                                JSONObject fullData = new JSONObject(response);
+                                JSONArray itemsArray = fullData.getJSONArray("items");
+                                for (int i = 0; i<itemsArray.length();i++){
+                                    JSONObject individualItem= itemsArray.getJSONObject(i);
+                                    String name = individualItem.getString("name");
+                                    walmartItems.add(new WalmartItem(name));
+                                }
+                                mRecyclerView = (RecyclerView)findViewById(R.id.recycler_view);
+
+                                LinearLayoutManager linearLayoutManager = new LinearLayoutManager(MainActivity.this, LinearLayoutManager.VERTICAL,false);
+                                mAdapter = new RecyclerWalmartAdapter(walmartItems);
```

Comment:
Don't recreate the whole adapter everytime a network call complete! This is very inefficient. Just set it up once, then call `notifyDataSetChanged()` after updating the data with the API results